### PR TITLE
chore: remove redundant push token code

### PR DESCRIPTION
**Pull request by johnxnguyen**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

We've changed the way we manage push tokens in the request strategy, namely we've moved away from the upstream object sync to using an action based sync. Because of this, we don't need to rely on the `UserClient` properties `pushToken` and `legacyPushToken` for the `PushTokenStrategy` to work. In fact, we don't even need to store the push token in the client, since it only affects the self client, as if can be stored in a single location such as the user defaults.

For this reason, we are removing the push token storage entirely from the user client, and stripping away the unnecessary properties and methods of `PushToken` that are no longer needed. **However we do not delete the `pushToken` property from the database, as we need it to migrate the token to user defaults.** We can remove it after some time.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Source/Model/UserClient/UserClient.swift [hunk 1]
```diff
@@ -119,28 +126,14 @@ public class UserClient: ZMManagedObject, UserClientType {
...
-                primitiveLegacyPushToken = try? JSONEncoder().encode(newValue)
-                self.didChangeValue(forKey: Keys.LegacyPushToken)
-                setLocallyModifiedKeys([Keys.LegacyPushToken])
-            }
-        }
+    /// Fetches and removes the old push token from the self client.
+    ///
+    /// - returns: the legacy push token if it exists.
+
+    public func retrieveLegacyPushToken() -> PushToken? {
```
Comment (KaterinaWire) [inline Source/Model/UserClient/UserClient.swift:133]: We should also have a public setter to nil

**Review by KaterinaWire — APPROVED** [review]

**Comment by KaterinaWire** [comment]
release major

**Comment by KaterinaWire** [comment]
release major

**Comment by zenkins** [comment]
Hello, @KaterinaWire! Merging PR for command **release major** for PR #**1363**. Will create a **major** release after merged.
Next version is 288.0.0.

**Pull request merged**